### ISSUE TITLE
Directives missing in the autogenerated GraphQL Schema if using code-first approach (annotations inside the model are ignored)

### ISSUE THREAD

**Issue by keadex** [nestjs/graphql#1597]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
I'm trying to build multiple GraphQL Microservices (Experience Service and Company Service) by using **Apollo Federation** and **code-first approach**.
I've followed this documentation: https://docs.nestjs.com/graphql/federation#federation

Basically:
   - inside the **Experience Service** you can find model, resolver and module related to the **Experience** entity.
   - inside the **Company Service** you can find model, resolver and module related to the **Company** entity. Moreover, in order to separate by concern, in the **Company Service** you can find an extension of the **Experience** model in order to add the _company_ field (as described here: https://docs.nestjs.com/graphql/federation#code-first-1):

https://github.com/keadex/keadex-einaudi/blob/0.0.1/apps/company/src/models/experience.model.ts
```
@Schema()
@ObjectType()
@Directive('@extends')
@Directive('@key(fields: "_id")')
export class Experience {
  @Field(() => String)
  @Directive('@external')
  _id: MongooseSchema.Types.ObjectId;

  @Field(() => [Company])
  @Prop({ type: [MongooseSchema.Types.ObjectId], ref: Company.name })
  companies: MongooseSchema.Types.ObjectId[] | Company[];
}
```

The problem is that, the autogenerated GraphQL Schema ignores all the Directives, included the @extends one:

```
# ------------------------------------------------------
# THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
# ------------------------------------------------------

directive @extends on INTERFACE | OBJECT

directive @external on FIELD_DEFINITION | OBJECT

directive @key(fields: String!) on INTERFACE | OBJECT

directive @provides(fields: String!) on FIELD_DEFINITION

directive @requires(fields: String!) on FIELD_DEFINITION
...
type Experience {
  _id: String!
  ciao: String!
  companies: [Company!]!
}
...
```

## Expected behavior
With the code-first approach I expect that NestJS, during the generation of the GraphQL Schema, takes into account all the directives added inside the model, autogenerating so a schema like the following one:

```
extend type User @key(fields: "id") {
  id: ID! @external
  posts: [Post]
}
```

## Minimal reproduction of the problem with instructions
The repo with the full code (Experience Service, Company Service, Gateway) is the following one: https://github.com/keadex/keadex-einaudi/tree/0.0.1

This is the Experience model (which extends the one inside the Experience Service - https://github.com/keadex/keadex-einaudi/blob/0.0.1/apps/experience/src/models/experience.model.ts) inside the Company Service: https://github.com/keadex/keadex-einaudi/blob/0.0.1/apps/company/src/models/experience.model.ts

## What is the motivation / use case for changing the behavior?
Is not possible to extend GraphQL types and create GraphQL microservices separated by concern leveraging on Apollo Federation. 

## Environment

<pre><code>
Nest version: 7.6.0

 
For Tooling issues:
- Node version: 12.13.0  
- Platform: Windows

**Comment by jmcdo29** [comment]
Can you make a minimum reproduction that doesn't have the database dependency? Is this only for federation or does it happen for regular GQL too?

**Comment by keadex** [comment]
I created two branches:
- https://github.com/keadex/keadex-einaudi/tree/no-db-depedency --> removed db dependency for the Company Service (https://github.com/keadex/keadex-einaudi/tree/no-db-depedency/apps/company)
- https://github.com/keadex/keadex-einaudi/tree/no-federation --> removed the federation for the Company Service (https://github.com/keadex/keadex-einaudi/tree/no-federation/apps/company) but the problem happens too

**Comment by kamilmysliwiec** [comment]
https://github.com/nestjs/graphql/issues/1278#issuecomment-744287059
https://github.com/nestjs/graphql/issues/1278#issuecomment-744287059
and the original issue https://github.com/graphql/graphql-js/issues/1343

**Comment by keadex** [comment]
After some investigation and reverse engineering, I think that the issue reported by @kamilmysliwiec (graphql/graphql-js#1343) has been fixed by an own implementation of Apollo.

In fact by inspecting nestjs code, the schema is autogenerated by using the `printSchema` utility from the `graphql` package (https://graphql.org/graphql-js/utilities/#printschema):
https://github.com/nestjs/graphql/blob/3a7468e865472bad7d9ddedc8cc7370c3b9e91a4/lib/graphql-schema.builder.ts#L106

By logging the `fileContent` variable, I can see exactly the (not correct - without directives) autogenerated file schema.

As stated above, Apollo has fixed the issue graphql/graphql-js#1343 in its own implementation, as also stated at the head of the code: https://github.com/apollographql/federation/blob/main/federation-js/src/service/printFederatedSchema.ts

<img src="https://user-images.githubusercontent.com/12099977/124369593-78226c80-dc6d-11eb-8ff1-8f82f1d63f3b.png" width="400">

In fact, nestjs uses that at runtime and by logging the generated schema by Apollo, I can see correctly the directives:

<img src="https://user-images.githubusercontent.com/12099977/124369630-03036700-dc6e-11eb-8824-4d6374787cef.png" width="400">

So in order to autogenerate from nestjs the correct GraphQL Schema (with the directives), in the following file I've changed the import of the `printSchema` function from `graphql` to `@apollo/federation` package:
https://github.com/apollographql/federation/blob/main/federation-js/src/service/printFederatedSchema.ts

By doing this, now the autogenerated GraphQL Schema file is correct (with the directives):

```
# ------------------------------------------------------
# THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
# ------------------------------------------------------
...
type Experience @extends @key(fields: "_id") {
  _id: String! @external
  companies: [Company!]!
}
...
```

**Comment by Kalmac** [comment]
@keadex , your solution is perfect ! 

Thanks a lot.
